clang/tools/driver/CMakeLists.txt

```diff
@@ -124,12 +124,19 @@ endif()
 set(features_file_src "${CMAKE_CURRENT_SOURCE_DIR}/features.json")
 set(features_file_dest "${CMAKE_BINARY_DIR}/share/clang/features.json")
 
+# Create list of preprocessor defines.
+add_custom_target(clang-features-file-dummy)
+llvm_update_compile_flags(clang-features-file-dummy)
+get_target_property(features_file_definitions clang-features-file-dummy COMPILE_DEFINITIONS)
+list(TRANSFORM features_file_definitions PREPEND -D)
+
 add_custom_command(OUTPUT ${features_file_dest}
                    COMMAND ${CMAKE_COMMAND} -E make_directory
                      ${CMAKE_BINARY_DIR}/share/clang
-                   COMMAND ${CMAKE_COMMAND} -E copy
+                   COMMAND ${CMAKE_C_COMPILER} -E -P -x c
```

Comment:
This doesn't take into account the spelling for different compilers.  Not all compilers use `-E`, `-P`, `-x` for these operations.  I think that we should use cmake to substitute the parameters using `configure_file`.

Reply:
Thanks for pointing that out and for the suggestion! @bnbarham noted that with `configure_file`, we'd need to create a CMake variable that ends with the trailing comma, which complicates things a bit.

I tried adding a simple executable target we can invoke to generate the `.json` file. This still gives us the power of preprocessor and correctly set up defines.

Reply:
Hmm, please make sure that you update compnerd/swift-build (for the Windows release builds) if you are going down that path.  The custom tool complicates cross-compiling, we now need to work with an additional custom tool that needs to be built for the build host.

Is there a reason that this cannot be done via CMake?  Ensuring that the string ends with a comma is pretty straightforward.

```
string(REGEX MATCH ",$" _has_trailing_comma "${JSON_FRAGMENT}")
if(NOT ${_has_trailing_comma})
  string(APPEND JSON_FRAGMENT ",")
endif()
```

Reply:
I see, that sounds like an unnecessary complication indeed. I switched to using `configure_file`.